microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/YAMLConfigSource.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.core.project;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+
+import org.eclipse.jdt.core.IJavaProject;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * YAML config source implementation
+ * 
+ * @author dakwon
+ *
+ */
+public class YAMLConfigSource extends AbstractConfigSource<LinkedHashMap<String, Object>> {
```

Comment:
Use Map (which is an interface) instead of LinkedHashMap (which is an implementation)